Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,934 @@
+      {
+        popUpDiv.css('top', 0);
+      }
+      else if (space_top > popUpDiv.height())
+      {
+        var bottom = Math.max(0, $(window).height() - referenceElement.offset().top - (referenceElement.outerHeight() - referenceElement.height()));
+
+        popUpDiv.css('top', 'auto');
+        popUpDiv.css('bottom', bottom);
+      }
+      else
+      {
+        popUpDiv.css('top', 'auto');
+        popUpDiv.css('bottom', 0);
+      }
+    }
+  }
+
+  public static ClosePopUp (updateFocus: boolean): void
+  {
+    if (DropDownMenu._blurTimer !== null)
+      clearTimeout(DropDownMenu._blurTimer);
+
+    if (DropDownMenu._currentPopup !== null
+        && document.body.contains(DropDownMenu._currentPopup))
+    {
+      const menuPopup = DropDownMenu._currentPopup;
+      const $menuButton = $('a[aria-controls="' + menuPopup.id + '"]');
+      $menuButton.unbind('focus', DropDownMenu.BlurHandler);
+      $menuButton.unbind('blur', DropDownMenu.BlurHandler);
+      const menuButton = $menuButton[0];
+      menuButton.setAttribute('aria-expanded', 'false');
+      menuButton.removeAttribute('aria-controls');
+      if (updateFocus === DropDownMenu._updateFocus)
+        menuButton.focus();
+      menuPopup.parentElement!.removeChild(menuPopup);
+    }
+
+    if (DropDownMenu._currentStatusPopup !== null
+        && document.body.contains(DropDownMenu._currentStatusPopup))
+    {
+      const statusPopup = DropDownMenu._currentStatusPopup;
+      // Clear the role=alert before removing to item to prevent screenreaders (JAWS) from announcing the old value during removal.
+      statusPopup.removeAttribute('role');
+      statusPopup.parentElement!.removeChild(statusPopup);
+    }
+
+    DropDownMenu._currentPopup = null;
+    DropDownMenu._currentStatusPopup = null;
+    DropDownMenu._currentMenu = null;
+  }
+
+  private static CreateItem(itemInfo: Nullable<DropDownMenu_ItemInfo>, selectionCount: number): Nullable<HTMLLIElement>
+  {
+    if (itemInfo == null)
+      return null;
+
+    var item;
+    if (itemInfo.Text == '-')
+      item = DropDownMenu.CreateSeparatorItem();
+    else
+      item = DropDownMenu.CreateTextItem(itemInfo, selectionCount);
+
+    return item;
+  }
+
+  private static CreateTextItem(itemInfo: DropDownMenu_ItemInfo, selectionCount: number): HTMLLIElement
+  {
+    var isEnabled = DropDownMenu.GetItemEnabled(itemInfo, selectionCount);
+
+    var item = document.createElement("li");
+
+    var className = DropDownMenu._itemClassName;
+    if (!isEnabled)
+      className = DropDownMenu._itemDisabledClassName;
+
+    item.className = className;
+    item.setAttribute('role', 'none');
+
+    var anchor = document.createElement("a");
+    anchor.setAttribute('role', 'menuitem');
+    anchor.setAttribute('tabindex', '-1');
+    if (isEnabled)
+    {
+      anchor.setAttribute('href', '#');
+    }
+    else
+    {
+      anchor.setAttribute('aria-disabled', 'true');
+    }
+    // TODO: missing nullcheck?
```

Comment:
See earlier comment regarding _itemClickHandler. It cannot be null once we're done.